smt.go

```diff
@@ -377,6 +377,106 @@ func (smt *SMT) Prove(key []byte) (proof *SparseMerkleProof, err error) {
 	return proof, nil
 }
 
+// ProveClosest generates a SparseMerkleProof of inclusion for the key
+// with the most common bits as the path provided
+func (smt *SMT) ProveClosest(path []byte) (
+	closestPath, closestValueHash []byte, // the closest leaf info for the key provided
+	proof *SparseMerkleProof, // proof of the key-value pair found
+	err error, // the error value encountered
+) {
+	var siblings []treeNode
+	var sib treeNode
+	var parent treeNode
+	var parentDepth int
+
+	node := smt.tree
+	depth := 0
+	for depth < smt.depth() {
```

Comment:
Request: can you add some comments throughout the business logic? 

You have this in the lower half of the function.

For example, above the `if node == nil {`  could be a good place, amongst others.